homeassistant/components/history/__init__.py

```diff
@@ -120,6 +120,11 @@ async def get(
         entity_ids = None
         if entity_ids_str:
             entity_ids = entity_ids_str.lower().split(",")
+            for entity_id in entity_ids:
+                if not valid_entity_id(entity_id):
```

Comment:
Please check if it's in the state machine first before calling this check as it can get expensive if the entity id list is large

Also this needs a test.

Thanks